plugins/content/iwcsocialmeta/iwcsocialmeta.php

```diff
@@ -0,0 +1,220 @@
+		if (!in_array($context, array('com_content.article')))
+		{
+			return true;
+		}
+
+		$app = JFactory::getApplication();
+
+		if ($app->isAdmin())
+		{
+			$str = (explode(".", $context));
+
+			if (!empty($data))
+			{
+				$db    = JFactory::getDbo();
+				$query = $db->getQuery(true);
+
+				$query->select('m.*');
+				$query->from($db->quoteName('#__iwcsocialmeta', 'm'));
+				$query->where($db->quoteName('m.article_id') . ' = ' . (int) $data->id);
+				$db->setQuery($query);
+
+				$results = $db->loadObjectList();
+
+				if ( !empty($results) )
+				{
+					$data->social_allowed    = explode(",", $results[0]->social_allowed);
+					$data->social_meta_title = $results[0]->social_meta_title;
+					$data->social_meta_desc  = $results[0]->social_meta_desc;
+					$data->social_meta_type  = $results[0]->social_meta_type;
+					$data->social_meta_image = $results[0]->social_meta_image;
+				}
+				else
+				{
+					$intro_image = $data->images['image_intro'];
+					$full_image  = $data->images['image_fulltext'];
+
+					if ($intro_image !== '' || $full_image !== '' )
+					{
+						if ($intro_image !== '' )
+						{
+							$data->social_meta_image = $intro_image;
+						}
+						else
+						{
+							$data->social_meta_image = $full_image;
+						}
+					}
+
+					$data->social_meta_title = $data->title;
+					$data->social_meta_desc  = $data->title;
+					$data->social_meta_type  = $str[1];
+				}
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Adds additional fields to the editing form
+	 *
+	 * @param   JForm  $form  The form to be altered.
+	 * @param   mixed  $data  The associated data for the form.
+	 *
+	 * @return  boolean
+	 *
+	 * @since   3.6
+	 */
+	public function onContentPrepareForm($form, $data)
+	{
+		if (!($form instanceof JForm))
+		{
+			$this->_subject->setError('JERROR_NOT_A_FORM');
+
+			return false;
+		}
+
+		// Check we are manipulating a valid form.
+		$name = $form->getName();
+		/*if (!in_array($name, array('com_content.article','com_contact.contact')))*/
+		if (!in_array($name, array('com_content.article')))
+		{
+			return true;
+		}
+
+		$app = JFactory::getApplication();
+
+		if ($app->isAdmin())
+		{
+			/* Add the fields to the form. */
+			JForm::addFormPath(__DIR__ . '/forms');
+			$form->loadFile('iwcsocialmeta', false);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Called right before the content is saved into the database
+	 *
+	 * @param   JForm    $context  The form to be altered.
+	 * @param   mixed    $article  The associated data for the form.
+	 * @param   boolean  $isNew    A boolean which is set to true if the content is about to be created.
+	 *
+	 * @return  boolean
+	 *
+	 * @since   3.6
+	 */
+	public function onContentBeforeSave($context,$article,$isNew)
+	{
+		$app     = JFactory::getApplication();
+		$input   = $app->input;
+		$session = JFactory::getSession();
+		$session->set("formData", $input->post->get('jform', array(), 'array'));
```

Comment:
Same here, `$app`,  `$input` and `$session` used only once.
**Final code should be like this**

```php
JFactory::getSession()->set(
	'formData',
	JFactory::getApplication()->input->post->getArray('jform', array())
);
```